community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/pipes/AllNodesScanPipe.scala

```diff
@@ -41,7 +41,7 @@ case class AllNodesScanPipe(ident: String)(val estimatedCardinality: Option[Doub
 
   override def monitor = pipeMonitor
 
-  override def localEffects: Effects = Effects(ReadsAllNodes)
+  override def localEffects: Effects = Effects(ReadsAllNodes).asLeafEffects
```

Comment:
We decided to move away from effects on pipes, right? If that's correct, remove this.